### DISCUSSION TITLE
Community meeting topics - Oct/Nov 2023

### DISCUSSION THREAD

**Discussion by Aggror** [stride3d/stride#1825] [General]
Hello there,

It has been quite some time since the last community meeting. In this topic I want to brainstorm about the topics that we can talk about. 
Keep in mind that this community meeting will be around 1 or  at max 2 hours. 

**What we will definitely talk about**
1. Summary of the Stride '**Contributor meeting'** (this meeting is purely meant for the core engine contributors to keep the chats focussed)
2. Current contributors overview, who is working on what and has the access rights to do so (code, website, docs, builds, official release, youtube, moderators etc)
3. Release of 4.2
4. Open Collective funding
5. Documentation and (Video)Tutorials
6. Community Toolkit
7. New Demo scene (2024 preparation)

**Other topic ideas**
- new contributors? 
- Special funding or plans: 
  - Plugins support
  - Demo scene 
  - Asset pipeline, multiple animations
  - Morph targets
  - https://opencollective.com/stride3d/projects 
- Official Nuget packages (code)
- Official open asset website (not an Asset store, purely a place for community made asset items)
- ......
- .......
- ......
- ......

**Comment by SeleDreams** [comment]
I think it would be important to talk about the aspects of stride to improve in order to improve the experience from new users.

**Comment by DaveMS** [comment]
I think it would be worth talking about marketing efforts. I've only heard about this engine in the last couple of days by chance, but it seems like a promising alternative for all the Unity refugees.

More Marketing -> Bigger community -> More donations -> Faster development.

**Reply by VaclavElias** [reply]
When I found Stride, I was looking specifically for C# Game Engine and it was easy to find for me as a C# Developer. The website seems to be SEO friendly for those phrases. How did you navigate to find Stride so we could assess the phrases if they are missing across our websites?

**Reply by ly3027929699** [reply]
When my browser searches for C# game engine, Xen appears instead of stride.

**Reply by ly3027929699** [reply]
We have many Unity developers here, but they know little about stride. Many people like gorgeous graphics, such as promotional scenes like those made by UE. This will be a very attractive place.

Although there are some template projects now, most people like to make judgments before downloading the software. Such judgments are inaccurate.

**Reply by Dreeheh** [reply]
Personally, I've found out about Stride engine by watching a video listing different alternatives to Unity

**Reply by DaveMS** [reply]
I found Stride on a reddit post listing Unity alternatives although I'm actually migrating from Unreal Engine myself.

**Reply by JeromyWalsh** [reply]
I found Stride through the .NET Foundation. https://dotnetfoundation.org/projects/current-projects

**Reply by Aggror** [reply]
@DaveMS Any tips on 'Do more marketing'? What would be something we could do to more marketing?

**Comment by Idomeneas1970** [comment]
Thanks for doing this @Aggror 
**Discussion Topic:** Creation of a flagship game (I am talking about a full game, illustrating the capabilities of the engine). Other game engines have done that I think. This will bring in revenue as well and it can be supported by the community. Have to figure out the details on how it can be accomplished without resorting to plagiarism.
**Discussion Topic:** Creation of a database for open source assets that can be included in the engine, including models, textures and animations. This will facilitate faster game creation and testing by users.
**Discussion Topic:** Creation of a website (or add to the manual) core engine concepts/entities and how can they be retrieved from script so that a user can find it fast. For example, if I want to grab a specific object at runtime, like a camera, have a single place where we can find commands like this:
To grab the camera object from within a script do this:
`CameraComponent camera = Game.Services.GetService<SceneSystem>().GraphicsCompositor.Cameras[0].Camera;
`
To grab a UI object from within a script do this:
```
            var page = Entity.Get<UIComponent>().Page;
            // Retrieve UI elements by Type and name
            PreviewImage = page.RootElement.FindVisualChildOfType<ImageElement>("PreviewImageHolder");
```
This will help users finding what they want fast. I realize that going through all samples you can figure this out eventually (like I did), but it will help to have it. And some of these are in the help but they are all spread out.
Thanks

**Reply by VaclavElias** [reply]
**Regarding the docs:** We have new docs in staging https://stride-doc-staging.azurewebsites.net/latest/en/ which we would like to release soon and then start working on the manual, including C# snippets improvements. Community could help with updating existing docs, with these, or just creating an issue in Docs, adding a snippet and which page should it be placed would be helpful, so others could place it there.

**Reply by Aggror** [reply]
Hi @Idomeneas1970 

1. That is somewhat what the official topic 'New Demo scene (2024 preparation)' is about. It will not be an entire flagship as that way to massive, but rather a demo scene showing of Stride's core features.
2. Open Asset collection for Stride (not directly an asset store), purely an open asset collection. Will add it to the list.
3. your 3rd suggestion is what topic 'Community Toolkit' is about. Communtiy toolkit contains many user extensions, examples and code only approaches. It will be part of the official stride repo's.

**Comment by ly3027929699** [comment]
Sorry. I'm looking forward to the new features of .net8, and the editor extension (plugin) system I'm used to in Unity, which I can barely live without. Also, have you considered exposing part of the editor API for game developers, this would make it easier to make some editor tools.

**Reply by Aggror** [reply]
It unfortunately is not as simple as 'exposing the editor'. Ideas for plugins and editor extensibility have been around for years, but have yet to bear fruit. With the new blood coming in to our community we are going to look into what is feasible.

See this proposal: https://github.com/stride3d/stride/issues/1120
and a recent ticket: https://github.com/stride3d/stride/discussions/1823

**Comment by Tonedus** [comment]
Such a meeting would definitely help to find people willing to participate in the creation of the demo or give interesting ideas for it.

I also noticed that the attention of the gamedev community passes by Stride, this is due to its recognizability. It might be worth creating a special presentation for indie studios and those looking for an easy to learn game engine. This information can be posted on various gamedev portals. This issue needs to be discussed.

**Reply by Aggror** [reply]
Wouldn't a nice demo be ideal for this? Having specific presentation for a various studios, would be very time consuming.

**Comment by JeromyWalsh** [comment]
As a current Unity user whose sole development environment is Linux, the lack of a Linux Editor & runtime support is a blocker. If possible, it would be good to discuss plans for Linux availability.

**Reply by Aggror** [reply]
We would definitely like to have it, but the rewrite to another platform is no small task.

**Reply by JeromyWalsh** [reply]
I completely understand. I'm happy to take the work on as it's blocking me from using Stride at the moment.

**Comment by SeleDreams** [comment]
It's a project that's definitely planned. Some members are thinking about
tackling the port to the Avalonia UI framework

Le mar. 26 sept. 2023 à 19:15, JeromyWalsh ***@***.***> a
écrit :

**Comment by JeromyWalsh** [comment]
I was looking at the contributor guidelines and it said to put your name on an issue if you wanted to work on something, but there are over 400 issues there now, some of which date back to 2018 and are still unresolved (but maybe completed).

Is there a roadmap some place that indicates the priority of work that needs to be done? I'm hoping there's one already, or if not, a plan to create one to help align all the new developers towards a common set of goals rather than having us all pick & choose our pet projects to work on. Not that that isn't fun. 🙂

Put differently, is there a single source of truth over what the developers *want* done, rather than a list of things that *can* be done?

**Reply by Agoxandr** [reply]
I would take a look at the open collective page. Those seem to be the most urgent.

**Comment by manio143** [comment]
Adding collective page for https://github.com/stride3d/stride/issues/70

**Comment by Kryptos-FR** [comment]
Hello.

Where is the meeting taking place?
I couldn't find that information anywhere.

**Reply by VaclavElias** [reply]
<img width="257" alt="image" src="https://github.com/stride3d/stride/assets/4528464/0e143115-41db-4867-8482-9773a960bb03">

**Reply by VaclavElias** [reply]
And don't worry, I was also looking where it is 🙂
![image](https://github.com/stride3d/stride/assets/4528464/9a8442d7-d765-42d5-8aa2-4bb51a50c624)
